spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractMessageChannelBinder.java

```diff
@@ -259,12 +259,19 @@ else if (e instanceof ProvisioningException) {
 		}
 		this.postProcessOutputChannel(outputChannel, producerProperties);
 
-		((SubscribableChannel) outputChannel)
-				.subscribe(new SendingHandler(producerMessageHandler,
-						HeaderMode.embeddedHeaders
-								.equals(producerProperties.getHeaderMode()),
-						this.headersToEmbed, useNativeEncoding(producerProperties)));
+		ReactiveStreamsConsumer[] reactiveStreamsConsumer = new ReactiveStreamsConsumer[1];
```

Comment:
That's strange...
Any explanations, please?
Perhaps you are facing a `final` error for that `afterUnbind()` functionality...
Consider to extract extra variable before that `binding` definition.
Another solution is to have an `AtomicReference<ReactiveStreamsConsumer>`.
But array logic is hard to digest. 😄

Reply:
Ok, I will go with the `AtomicReference`.